classes/core.php

```diff
@@ -1390,6 +1392,85 @@ static public function akismet_scanner($config, $form)
 
 	}
 
+	/**
+	 * Catch known functions that were supported at https://calderaforms.com/doc/calculation-fields/ and are not sipported by parser https://github.com/denissimon/formula-parser
+	 *
+	 * @since 1.8.5
+	 * 
+	 * @param string $formula The string used to evaluate the math formula
+	 *
+	 * @return bool||string of coma seperated deprecated symbol
+	 */
+	static public function check_deprecated_math_functions($formula)
+	{	
+		//List known depracated functions
+		$deprecated = array( ',', '**', 'pow', 'acos', 'asin', 'atan', 'atan2', 'ceil', 'floor', 'max', 'min', 'random', 'round');
+		//Look for matching known symbol in formula and push it in a symbols array
+		foreach ($deprecated as $symbol) {
+			if (strpos($formula, $symbol) !== FALSE) { 
+				$symbols[] = $symbol;
+			}
+		}
+		//If symbols array is not empty convert it to a string and return the string
+		if(is_array($symbols) && !empty($symbols)){
+			$symbols = implode( ',', $symbols);
+			return $symbols;
+		} else {
+			//Else return false
+			return false;
+		}
+		
+	}
+
+	/**
+	 * Calculation based on create_function() ( deprecated since PHP 7.2.0 )
+	 * 
+	 * @since 1.8.5
+	 * 
+	 * @param string $formula that needs to be parsed and processed as a calculation
+	 * 
+	 * @return result of calculation
+	 */
+	static public function original_calculation_job( $formula )
+	{
+		$total_function = create_function(null, 'return ' . $formula . ';');
+		$total = $total_function();
+
+		return $total;
+	}
+
+	/**
+	 * Process when php >= 7.2.0 and deprecated is caught
+	 * 
+	 * @since 1.8.5
+	 * 
+	 * @param string $formula to be sent to original calculation job
+	 * @param string $deprecated function or symbol caught
+	 * 
+	 * @return original process and notice for admin users
+	 */
+	static public function calculation_deprecated_caught_process( $formula, $deprecated )
+	{
+		if( current_user_can( Caldera_Forms::get_manage_cap('admin') ) ) {
+			
+			add_filter('caldera_forms_render_notices', function( $out ) use ($deprecated){
+
+				$message = sprintf(
+					'"%s" %s',
+					$deprecated,
+					__('has been deprecated in calculation fields for compatibility with PHP 7.2 or later', 'caldera-forms')
+				);
+
+				$out['error']['note'] = $message;
+
+				return $out;
+
+			}, 10);
```

Comment:
Arguably, this should have a higher priority. We use 10 in a few places https://github.com/CalderaWP/Caldera-Forms/search?q=caldera_forms_ajax_return&unscoped_q=caldera_forms_ajax_return

Moving it to 11 would make it over ride CF Pro notices, which is good. So maybe set to 25, but not 99.